### PR TITLE
[FIX] account_edi: correct fix about multiple docs per cancellation

### PR DESCRIPTION
**Pull request by jco-odoo**
In commit b092cf349fa87639485a724ecd7b33c4c81d97c0 we checked if the other documents were also in the cancelled state, but the problem is that this way if you cancel multiple invoices and only one is in that case it will apply to all of them.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by jco-odoo** [comment]
@william-andre I tested that the cancellation for Peru still works (just to be sure the code would not have any side effects)

File: addons/account_edi/models/account_edi_document.py [hunk 1]
```diff
@@ -159,7 +159,8 @@ def _postprocess_cancel_edi_results(documents, edi_result):
                         'blocking_level': False,
                     })
 
-                    if move.state == 'posted':
+                    if move.state == 'posted' and all(doc.state == 'cancelled' or not doc.edi_format_id._needs_web_services()
+                                                      for doc in move.edi_document_ids):
```
Comment (william-andre) [inline addons/account_edi/models/account_edi_document.py:163]:
```suggestion
                    if move.state == 'posted' and all(
                        doc.state == 'cancelled' 
                        or not doc.edi_format_id._needs_web_services()
                        for doc in move.edi_document_ids
                    ):
```

**Review by william-andre — COMMENTED** [review]
Makes sense but could be more readable :wink:

File: addons/account_edi/models/account_edi_document.py (hunk 1 above)
Comment (jco-odoo) [reply addons/account_edi/models/account_edi_document.py]: adapted.

**Review by william-andre — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**